OneCommunityGlobal/HighestGoodNetworkApp
# fix timelog permission problem after improving performance

### Description
This is an updated version of a previous merged and reverted  PR #762 : fix the bug that when an admin view other user's dashboard, they do not get the permission to add time entry for that user (the orange add button doesn't show).

### Mainly changes explained:
Please check the #762  first to get the idea of what most of the code changes were for.
base on that, a replacement of the first prop for the `hasPermission` function (`auth.user`-->`auth.user.role`) will fix the bug, because the latter one is the correct datatype that is expected.

### How to test (this bug fix):
Log in as an admin-->
on the leaderboard, click the dot before a user's name to open their dashboard-->
see the add time entry button on that user's dashboard-->
add a time entry for them and save successfully.
<img width="570" alt="img" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/d50ae130-875b-4531-8f4d-36aa459e2be9">
